src/core/mavlink_parameters.h

```diff
@@ -546,6 +552,12 @@ class MAVLinkParameters {
     std::mutex _all_param_mutex{};
 
     // dl_time_t _last_request_time = {};
+
+    std::map<std::string, ParamValue> _param_server_store;
+    void process_param_request_read(const mavlink_message_t& message);
+    void process_param_ext_request_read(const mavlink_message_t& message);
+    void process_param_request_list(const mavlink_message_t& message);
+    void process_param_set(const mavlink_message_t& message);
```

Comment:
My main question here is whether it makes sense to add this functionality to the existing `MAVLinkParameters` class, or a separate class. To me it's not entirely clear what is client and what is server related in this class. If it's mostly separate except from the type it might make sense to have a separate class and only re-use some of the types and helper functions.

Reply:
I agree this should be separated for clarity. With provide/retrieve naming it is perhaps a little more clear. For now it will require re-implementation of the param worker/jobs so we can do this later.